nox/popen.py

```diff
@@ -21,11 +21,12 @@
 from typing import IO, Mapping, Sequence
 
 
-def shutdown_process(proc: subprocess.Popen) -> tuple[bytes, bytes]:
+def shutdown_process(
+    proc: subprocess.Popen, child_timeout: float | None
+) -> tuple[bytes, bytes]:
     """Gracefully shutdown a child process."""
-
     with contextlib.suppress(subprocess.TimeoutExpired):
-        return proc.communicate(timeout=0.3)
+        return proc.communicate(timeout=child_timeout or 0.3)
```

Comment:
Should this timeout (the first one) be the one controlled by the variable or should it be the second? I'm on the fence.